### ISSUE TITLE
Can't validate domain name without using public key nor pinned certificates?

### ISSUE THREAD

**Issue by jcayzac** [AFNetworking/AFNetworking#2560]
In AFSecurityPolicy:

```objc
    NSMutableArray *policies = [NSMutableArray array];
    if (self.validatesDomainName) {
        [policies addObject:(__bridge_transfer id)SecPolicyCreateSSL(true, (__bridge CFStringRef)domain)];
    }
    [...]
    if (self.SSLPinningMode != AFSSLPinningModeNone && !AFServerTrustIsValid(serverTrust) && !self.allowInvalidCertificates) {
        return NO;
    }
    [...]
    switch (self.SSLPinningMode) {
        case AFSSLPinningModeNone:
            return YES; <<< ?
```

…why do you always return `YES` even if `validatesDomainName` is set? IMHO this last line should read 
`return AFServerTrustIsValid(serverTrust)`, with `serverTrust` only containing the domain policy set above.

**Comment by infimache** [comment]
I have similar issue with current implementation of <code>-evaluateServerTrust:forDomain:</code>

When <code>SSLPinningMode == AFSSLPinningModeNone</code>, no validation of server trust is performed and <code>-evaluateServerTrust:forDomain:</code> always returns <code>YES</code>.

I believe issue was introduced by commit <link> https://github.com/AFNetworking/AFNetworking/commit/104ce04105098466ea0ea4e337af554d7b9df195</link>.
It states that change was performed due to failing test, but for me tests succeed with or without the change.

Proposed solution is to revert mentioned commit.

**Comment by jcayzac** [comment]
I agree 104ce04 looks like a bad solution for the test it was supposed to fix.

**Comment by jcayzac** [comment]
Note that the security implication for this is that by default AFNetworking happily connects to https servers with invalid certificates. The company I work for has several dozen apps that just had to be updated to work around this ーit's an e-commerce company, so a few people there were quite stressed as you can guess :dancers:

**Comment by mattt** [comment]
Fixed by 90bd342774b312c86352703e04969c561643c99e. See #2611 for discussion.

**Comment by peterpaulis** [comment]
to go around this add

<code>
        // security policy
        {
            AFSecurityPolicy* policy = [AFSecurityPolicy policyWithPinningMode:AFSSLPinningModeNone];
            [policy setValidatesDomainName:NO];
            [policy setAllowInvalidCertificates:!SERVICE_FORCE_HTTPS];
            self.requestOperationManager.securityPolicy = policy;
        }
</code>
